frontend/src/components/Division.tsx

```diff
@@ -0,0 +1,187 @@
+  { key: 2, value: i18n.t('division:role') },
+];
+
+export const DivisionTableHeader = [
+  {
+    id: 'name',
+    name: i18n.t('general:name'),
+    sortable: true,
+    grow: 2,
+    minWidth: '15rem',
+    selector: (row: IDivision) => row.name,
+  },
+  {
+    id: 'membersCount',
+    name: i18n.t('division:members'),
+    sortable: true,
+    grow: 2,
+    minWidth: '15rem',
+    selector: (row: IDivision) => row.membersCount,
+  },
+  {
+    id: 'createdBy.name',
+    name: i18n.t('general:created_by'),
+    sortable: true,
+    grow: 2,
+    minWidth: '15rem',
+    selector: (row: IDivision) => row.createdBy.name,
+  },
+  {
+    id: 'createdOn',
+    name: i18n.t('general:created_on'),
+    sortable: true,
+    grow: 2,
+    minWidth: '15rem',
+    selector: (row: IDivision) => formatDate(row.createdOn),
+  },
+];
+
+interface DivisionProps {
+  data: IPaginatedEntity<IDivision> | undefined;
+  isLoading: boolean;
+  divisionType: DivisionType;
+  page: number;
+  onSort: (column: TableColumn<IDivision>, direction: SortOrder) => void;
+  onChangePage: (newPage: number) => void;
+  onRowsPerPageChange: (rows: number) => void;
+  onTabChange: (id: number) => void;
+}
+
+const Division = ({
+  data,
+  isLoading,
+  divisionType,
+  page,
+  onSort,
+  onChangePage,
+  onRowsPerPageChange,
+  onTabChange,
+}: DivisionProps) => {
+  // component actions
+  const onAdd = () => {
+    alert('Not yet implemented');
+  };
+
+  // row actions
+  const onView = (row: IDivision) => {
+    alert(`Not yet implemented, ${row}`);
+  };
+
+  const onEdit = (row: IDivision) => {
+    alert(`Not yet implemented, ${row}`);
+  };
+
+  const onDelete = (row: IDivision) => {
+    alert(`Not yet implemented, ${row}`);
+  };
+
+  // menu items
+  const buildDivisionActionColumn = (): TableColumn<IDivision> => {
+    const divisionMenuItems = [
+      {
+        label: i18n.t('general:view'),
+        icon: <EyeIcon className="menu-icon" />,
+        onClick: onView,
+      },
+      {
+        label: i18n.t('general:edit'),
+        icon: <PencilIcon className="menu-icon" />,
+        onClick: onEdit,
+      },
+      {
+        label: i18n.t('general:delete'),
+        icon: <TrashIcon className="menu-icon" />,
+        onClick: onDelete,
+        alert: true,
+      },
+    ];
+
+    return {
+      name: '',
+      cell: (row: IDivision) => <Popover row={row} items={divisionMenuItems} />,
+      width: '50px',
+      allowOverflow: true,
+    };
+  };
+
+  return (
+    <Tabs tabs={DivisionTabs} onClick={onTabChange}>
+      <Card>
+        <CardHeader>
+          <h3>{i18n.t(`${divisionType}`)}</h3>
+          <Button
+            className="btn-outline-secondary"
+            label={i18n.t('general:add')}
+            icon={<PlusIcon className="h-5 w-5" />}
+            onClick={onAdd}
+          />
+        </CardHeader>
+        <CardBody>
+          <DataTableComponent
+            className={classNames(
```

Comment:
Daca style-ul asta ar trebui tratat asa pentru fiecare table, are sens sa il avem tratat direct in DataTableComponent?

Reply:
Mutat in DataTableComponent.